src/components/elements/favorite-button.tsx

```diff
@@ -0,0 +1,32 @@
+"use client";
+import useFavorites, { Favorite } from "@lib/hooks/useFavorites";
+import { HTMLAttributes } from "react";
+import { HeartIcon } from "@heroicons/react/24/outline";
+import { clsx } from "clsx";
+import { useIsClient } from "usehooks-ts";
+
+type Props = HTMLAttributes<HTMLButtonElement> & Favorite
+
+const FavoriteButton = ({ uuid, title, path, units, ...props }: Props) => {
+  const { favs, addFav, removeFav } = useFavorites();
+  const isFavorite = favs.some((fav) => fav.uuid === uuid);
```

Comment:
hmm. i would've used a `!!favs.filter()` I wonder which is more efficient. :thinking: